File: Backend/app/schemas/shifts.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field

class ShiftBase(BaseModel):
    name: str = Field(..., description="Name of the shift (e.g. Morning, Afternoon)")
    start_time: str = Field(..., description="Start time in HH:MM format", pattern="^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$")
    end_time: str = Field(..., description="End time in HH:MM format", pattern="^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$")
    is_active: bool = True

class ShiftCreate(ShiftBase):
    pass

class ShiftUpdate(BaseModel):
    name: Optional[str] = None
    start_time: Optional[str] = Field(None, pattern="^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$")
    end_time: Optional[str] = Field(None, pattern="^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$")
    is_active: Optional[bool] = None

class ShiftResponse(ShiftBase):
    id: str
    created_at: datetime
    updated_at: datetime

File: Frontend/src/lib/api/shifts.ts
import { apiClient } from './client';

export interface Shift {
    id: string;
    name: string;
    start_time: string;
    end_time: string;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface ShiftCreate {
    name: string;
    start_time: string;
    end_time: string;
    is_active?: boolean;
}

export interface ShiftUpdate {
    name?: string;
    start_time?: string;
    end_time?: string;
    is_active?: boolean;
}

export const shiftsApi = {
    list: async (): Promise<Shift[]> => {
        return apiClient.get<Shift[]>('/shifts/');
    },

    create: async (data: ShiftCreate): Promise<Shift> => {
        return apiClient.post<Shift>('/shifts/', data);
    },

    update: async (id: string, data: ShiftUpdate): Promise<Shift> => {
        return apiClient.put<Shift>(`/shifts/${id}`, data);
    },

    delete: async (id: string): Promise<{ message: string }> => {
        return apiClient.delete<{ message: string }>(`/shifts/${id}`);
    }
};
